pymine_net/types/vector.py

```diff
@@ -1,30 +1,27 @@
-from typing import TypeVar
+from dataclasses import dataclass
+from typing import Generic, TypeVar
 
-T = TypeVar("T")
+T = TypeVar("T", bound=float)
 
 
-class Vector3:
+@dataclass(slots=True)
```

Comment:
The slots kwarg is is a 3.10+ feature iirc, please revert back to using a regular class.

Reply:
Can't we use `__slots__ = (...)` on dataclasses directly? Dataclasses themselves are 3.7+

Reply:
Can't we use `__slots__ = (...)` on dataclasses directly? Dataclasses themselves are 3.7+

Reply:
They kinda work: https://stackoverflow.com/questions/50180735/how-can-dataclasses-be-made-to-work-better-with-slots
But for now I'd rather just use

Reply:
I see, so that's probably not worth doing, however we could still consider using: [dataclassy library](https://github.com/biqqles/dataclassy) or [attrs library](https://pypi.org/project/attrs/) on which python's dataclasses were based on. Both of these support 3.7 and have `slots` feature.

### Examples of working slotted dataclasses in 3.7+

<details open>
 <summary>With <b><code>dataclassy</code></b></summary>
 
 ```py
 from dataclassy import dataclass
 
 @dataclass(slots=True)
 class Foo:
     x: int = 5
     y: str = "hi"
 ```
 </details>
 
 <details open>
 <summary>With <b><code>attrs</code></b></summary>
 
 ```py
 import attr
 
 @attr.s(slots=True)
 class Foo:
      x: int = 5
     y: str = "hi"
 ```
 
 </details>

Reply:
I think for now we can just stick with regular classes